docs/release-note-guide.md

```diff
@@ -0,0 +1,40 @@
+# Release Notes Language Style Guide
```

Comment:
modified from our the release notes guide of TiDB, which is not published because the original doc is only for TiDB maintainers